django/thunderstore/repository/mixins.py

```diff
@@ -1,7 +1,28 @@
+from django.utils.functional import cached_property
+
+from thunderstore.community.models.community import Community
+from thunderstore.community.models.community_site import CommunitySite
+
+
 class CommunityMixin:
-    @property
+    @cached_property
     def community_identifier(self):
         return (
             self.kwargs.get("community_identifier", None)
             or self.request.community.identifier
         )
+
+    @cached_property
+    def community(self):
+        return Community.objects.get(identifier=self.community_identifier)
+
+    @cached_property
+    def community_site(self):
+        return CommunitySite.objects.get(community=self.community)
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context["community_identifier"] = self.community_identifier
+        context["community"] = self.community
+        context["community_site"] = self.community_identifier
```

Comment:
Is this valid? It looks like it wouldn't be, but it also looks like it's unused if it hasn't caused issues..?

Reply:
Whoops, yeah that's invalid. Actually I wasn't supposed to add that yet.

Reply:
Added now this the next PR where it was supposed to be, also corrected to actually have the communitysite object in it.